cql3/statements/alter_keyspace_statement.cc

```diff
@@ -84,10 +84,21 @@ void cql3::statements::alter_keyspace_statement::validate(query_processor& qp, c
             auto new_ks = _attrs->as_ks_metadata_update(ks.metadata(), *qp.proxy().get_token_metadata_ptr(), qp.proxy().features());
 
             if (ks.get_replication_strategy().uses_tablets()) {
-                const std::map<sstring, sstring>& current_rfs = ks.metadata()->strategy_options();
-                for (const auto& [new_dc, new_rf] : _attrs->get_replication_options()) {
-                    auto it = current_rfs.find(new_dc);
-                    sstring old_rf = it != current_rfs.end() ? it->second : "0";
+                const std::map<sstring, sstring>& current_rf_per_dc = ks.metadata()->strategy_options();
+                auto new_rf_per_dc = _attrs->get_replication_options();
+                new_rf_per_dc.erase(ks_prop_defs::REPLICATION_STRATEGY_CLASS_KEY);
+                for (const auto& [new_dc, new_rf] : new_rf_per_dc) {
+                    sstring old_rf = "0";
+                    auto new_dc_in_current_mapping = current_rf_per_dc.find(new_dc);
+                    if (new_dc_in_current_mapping != current_rf_per_dc.end()) {
+                        old_rf = new_dc_in_current_mapping->second;
+                    } else if (qp.proxy().get_token_metadata_ptr()->get_topology().get_datacenters().count(new_dc) == 0) {
```

Comment:
`contains()` seems to be more appropriate than `count()` since you're not really interested in counting the occurrences in the the set, just if `new_dc` exists or not.

Reply:
Right, thanks. Somehow I incorrectly remembered that `contains()` is a c++23 feature, but [it's available since C++20](https://en.cppreference.com/w/cpp/container/map/contains). Hmm, I'm not even sure, but maybe we do support C++23 already?

Reply:
yes, it's available since c++20 and we use it in scylla for several years now.

Reply:
Fixed.